### PR TITLE
[skip ci] Added instuctions on building a docker image from dist

### PR DESCRIPTION
**Pull request by chupman**
I was building the container while reviewing a PR and thought it would be great to include instructions on how to use a release distribution since it can be a lot faster. 
I also brought in the update from this [commit](https://github.com/JanusGraph/janusgraph/commit/93ccb0dc10bc82ceee574a8887f2ecf6204bc3d4#diff-396d0b75211f5646b2fa730fbdf80a40) since I think it should have been targeted at the 0.2 branch originally.

Signed-off-by: Chris Hupman <chupman@us.ibm.com>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [X] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [X] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [X] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [X] Have you ensured that format looks appropriate for the output in which it is rendered?
- [X] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: BUILDING.md [hunk 1]
```diff
@@ -30,13 +30,32 @@ To build with only the TinkerPop tests:
...
+```
+mvn clean install -Pjanusgraph-release -Dgpg.skip=true -DskipTests=true
+```
+This command generates the distribution archive in `janusgraph-dist/janusgraph-dist-hadoop-2/target/janusgraph-$VERSION-hadoop2.zip`.
+For more details information, please see [here](janusgraph-dist/README.md#building-zip-archives)
+
 ## Building Docker Image for JanusGraph Gremlin Server
 
-To build and run Docker images with JanusGraph and Gremlin Server, configured 
+In order to build Docker image for JanusGraph Gremlin Server, a
```
Comment (FlorianHockmann) [inline BUILDING.md]: Please change this to _JanusGraph Server_ to avoid confusion as that's what we're using in the docs.

File: BUILDING.md [hunk 2]
```diff
@@ -30,13 +30,32 @@ To build with only the TinkerPop tests:
...
+refer to `To build the distribution archive` section to build the distribution 
+archive first. You can also use an [official release](https://github.com/JanusGraph/janusgraph/releases) to avoid building.
+To do so check out the release tag you wish to build, example: `git checkout v0.2.0`. Then create target
+directory that houses the distribution zip with `mkdir janusgraph-dist/janusgraph-dist-hadoop-2/target`.
+The [downloaded release](https://github.com/JanusGraph/janusgraph/releases) 
+is then placed in the recently created target directory. Note that if the
+tag is not found you can run `git fetch --all --tags --prune` and then rerun the checkout command.
+
+Once the distribution is in place use the following command
+to build and run Docker images with JanusGraph and Gremlin Server, configured
```
Comment (FlorianHockmann) [inline BUILDING.md]: Again, this should be just JanusGraph Server in my opinion.

**Comment by FlorianHockmann** [comment]
The changes look good, I just commented on two nits. Although I'm not sure whether someone wants to build a Docker image of an existing release once we have an official Docker image (janusgraph/janusgraph-docker#1). But we can still merge this for now and then decide later on what we do with these image in the main repo (like using it only for testing of other components).

File: BUILDING.md (hunk 1 above)
Comment (chupman) [reply BUILDING.md]: I pulled this section from master, and I agree it sounds better as JanusGraph Server. I updated line 33 in addition to the 2 you pointed out. Thanks for the review.

**Pull request merged**